File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
	if n <= 0 {
		return 0, errors.New("")
	}
	steps := 0
	for n != 1 {
		switch {
		case n%2 == 0:
			n = n / 2
		default:
			n = 3*n + 1
		}
		steps++
	}
	return steps, nil
}

File: solutions/zig/collatz-conjecture/1/collatz_conjecture.zig
// Please implement the `ComputationError.IllegalArgument` error.

pub const ComputationError = error{
    IllegalArgument,
};

pub fn steps(number: usize) anyerror!usize {
    if (number < 1) {
        return ComputationError.IllegalArgument;
    }

    var step: usize = 0;
    var res = number;

    while (res > 1) {
        step = step + 1;
        if (res % 2 == 0) {
            res = res / 2;
        } else {
            res = res * 3 + 1;
        }
    }

    return step;
}
